src/ssh/azext_ssh/_params.py

```diff
@@ -8,11 +8,14 @@ def load_arguments(self, _):
 
     with self.argument_context('ssh vm') as c:
         c.argument('vm_name', options_list=['--vm-name', '--name', '-n'], help='The name of the VM')
-        c.argument('ssh_ip', options_list=['--ip'], help='The public IP address (or hostname) of the VM')
+        c.argument('ssh_ip', options_list=['--ip', '--hostname'],
+                   help='The public (or reachable private) IP address (or hostname) of the VM')
         c.argument('public_key_file', options_list=['--public-key-file', '-p'], help='The RSA public key file path')
         c.argument('private_key_file', options_list=['--private-key-file', '-i'], help='The RSA private key file path')
         c.argument('use_private_ip', options_list=['--prefer-private-ip'],
-                   help='Will use a private IP if available. By default only public IPs are used.')
+                   help='Will prefer private IP.  Requires connectivity to the private IP.')
+        c.argument('port', options_list=['--port'], help='SSH port')
+        c.argument('ssh_args', options_list=['--ssh-args'], help='SSH Additional arguments')
```

Comment:
I don't think `--ssh-args` will work as expected, as it only takes exactly one value. The user may usually want to provide more than one value. (https://github.com/Azure/azure-cli/issues/19101)

Also, if not handled correctly, leading dashes in parameters like `--ssh-args -o SomeSshOption=3` will corrupt the parsing. See details at https://github.com/Azure/azure-cli/issues/18869.